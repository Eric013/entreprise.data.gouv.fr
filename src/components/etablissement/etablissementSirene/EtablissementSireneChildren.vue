<template>
  <div class="company__panel panel">
    <div v-if="isSiegeSocial" class="company__item">
      <div class="company__item-key">Cet établissement est le siège social</div>
    </div>
    <div v-if="!isSiegeSocial" class="company__item">Siège social :
      <router-link tag="div" class="company__item-link" :to="{ name: 'Etablissement', params: {searchId: resultSiegeSocial.siret}}">
        {{ resultSiegeSocial.nom_raison_sociale | removeExtraChars }}
      </router-link>
    </div>
    <div v-if="haveChildrenEtablissements" class="company__item lineup">
      <div class="company__item-key">Etablissements enfants : {{ totalResultsOtherSirens }} résultats ({{ maxChildrenEtablissements }} affichés)</div>
      <template v-if="thereAreMoreThanMaxChildren">
        <div class="company__item-link" v-show="!visibleChildren" @click="showAllChildren">Afficher la totalité</div>
        <div class="company__item-link" v-show="visibleChildren" @click="hideAllChildren">Réduire</div>
      </template>
      <ul>
        <router-link tag="li"
                      class="company__item-link"
                      :to="{ name: 'Etablissement', params: {searchId: siret}}"
                      v-for="siret in resultOtherSirens"
                      :key="siret">
          {{ siret }}
        </router-link>
      </ul>
    </div>
  </div>
</template>

<script>
import Constants from '@/constants.js'
import Filters from '@/components/mixins/filters'

export default {
  name: 'EtablissementSireneChildren',
  data () {
    return {
      maxChildrenEtablissementsToShow: Constants.ETABLISSEMENT_SHOW_MAX_CHILDREN_SIRETS,
      visibleChildren: false
    }
  },
  computed: {
    resultSirene () {
      return this.$store.getters.singlePageEtablissementSirene
    },
    resultSiegeSocial () {
      return this.$store.getters.storedSirenSiege || { siret: 0 }
    },
    resultOtherSirens () {
      const maxResults = this.maxChildrenEtablissementsToShow
      return this.$store.getters.storedSirenChildren.slice(0, this.maxChildrenEtablissementsToShow)
    },
    haveChildrenEtablissements () {
      return this.totalResultsOtherSirens >= 1
    },
    totalResultsOtherSirens () {
      // Total result minus Siege Social
      return this.$store.getters.storedSirenTotalResults - 1
    },
    maxChildrenEtablissements () {
      return Math.min(this.totalResultsOtherSirens, this.maxChildrenEtablissementsToShow)
    },
    isSiegeSocial () {
      if (!this.resultSirene) {
        return
      }
      if (this.resultSirene.is_siege == '1') {
        return true
      }
      return false
    },
    thereAreMoreThanMaxChildren () {
      return this.totalResultsOtherSirens >= this.maxChildrenEtablissementsToShow
    }
  },
  methods: {
    showAllChildren () {
      this.maxChildrenEtablissementsToShow = this.totalResultsOtherSirens
      this.visibleChildren = true
    },
    hideAllChildren () {
      this.maxChildrenEtablissementsToShow = Constants.ETABLISSEMENT_SHOW_MAX_CHILDREN_SIRETS
      this.visibleChildren = false
    }
  },
  mixins: [Filters]
}
</script>

<style lang="scss" scoped>
  .panel {
    // width: 100%;
    border: 2px solid $color-lighter-blue;
  }
</style>
