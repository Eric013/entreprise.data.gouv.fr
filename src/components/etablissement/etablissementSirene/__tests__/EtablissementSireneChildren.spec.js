import EtablissementSireneChildren from '@/components/etablissement/etablissementSirene/EtablissementSireneChildren.vue'
import { createLocalVue, shallow, mount, RouterLinkStub } from '@vue/test-utils'
import { __createMocks as createStoreMocks } from '@/store/index.js'
import Vuex from 'vuex'

const localVue = createLocalVue()
localVue.use(Vuex)
jest.mock('@/store/index.js')
jest.mock('@/router/index.js')
localVue.component('router-link', RouterLinkStub)

describe('EtablissementSireneChildren.vue', () => {
  let storeMocks
  let wrapperEPC
  let etablissementSireneChildren

  beforeEach(() => {
    storeMocks = createStoreMocks()

    wrapperEPC = shallow(EtablissementSireneChildren, {
      localVue,
      store: storeMocks.store
    })
    etablissementSireneChildren = wrapperEPC.vm
  })

  test('Computed value Result returns the right store getter', () => {
    expect(etablissementSireneChildren.resultSirene).toBe(storeMocks.store.getters.singlePageEtablissementSirene)
  })

  test('If company is social siege, the page display this information', () => {
    wrapperEPC.vm.resultSirene.is_siege = '1'
    wrapperEPC.update()
    expect(wrapperEPC.vm.isSiegeSocial).toBeTruthy()
    expect(wrapperEPC.find(".company__item-key").text()).toBe('Cet établissement est le siège social')
  })

  test('If company is not social siege, the page dont display this information', () => {
    wrapperEPC.vm.resultSirene.is_siege = '0'
    wrapperEPC.update()
    expect(wrapperEPC.vm.isSiegeSocial).toBeFalsy()
  })

  // TODO: Theses tests when router-link will be testable
  test('If company is not social siege, the page display a link to the social siege')
  test('clicking the link to social siege goes to the social siege page')
  test('If there are children etablissement, the page display them')
  test('Clicking on a children etablissement goes to its page')
  test('if less than Max Children etablissement, the page display all children')
  test('if more than Max Children etablissement, the page display only MaxChildren')
  test('clicking on showAllChidren displays all children')
  test('clicking on hideAllChidren goes back to Max Children etablissement displayed')
})

describe('EtablissementSireneChildren.vue snapshot', () => {
  localVue.component('router-link', RouterLinkStub)

  const storeMocks = createStoreMocks()
  const wrapperEPC = mount(EtablissementSireneChildren, {
    localVue,
    store: storeMocks.store
  })

  test('It match the snapshot', () => {
    expect(wrapperEPC.vm.$el).toMatchSnapshot()
  })
})