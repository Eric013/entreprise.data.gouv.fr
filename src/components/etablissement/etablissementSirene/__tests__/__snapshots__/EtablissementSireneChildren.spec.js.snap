// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EtablissementSireneChildren.vue snapshot It match the snapshot 1`] = `
<div
  class="company__panel panel"
>
  <!---->
   
  <div
    class="company__item"
  >
    Si√®ge social :
    
    <div
      class="company__item-link"
    >
      
      
    
    </div>
  </div>
   
  <!---->
</div>
`;
